### ISSUE TITLE
Inline plugin global styles

### ISSUE THREAD

**Issue by tylerforesthauser** [tw-in-js/twind#89]
So since the current approach to declaring global styles seems to be best accomplished through preflight, it would be nice to provide some kind of merging strategy for common selector keys as to avoid this sort of code...

```js
preflight: ({ html, ...preflight }, { theme }) => ({
  ...preflight,
  html: {
    color: theme('colors.gray.900'),
    ...html,
  },
})
```

Otherwise, I think there should be some discussion about ways to declare global styles.

Regarding this new (:global)[https://github.com/tw-in-js/twind/blob/main/docs/plugins.md#inject-global-styles] property, is it at all possible to declare global styles from within _inline_ plugins? That would be a nifty way to allow declaration of global styles from virtually anywhere in your app.

**Comment by sastan** [comment]
> Regarding this new (:global)[https://github.com/tw-in-js/twind/blob/main/docs/plugins.md#inject-global-styles] property, is it at all possible to declare global styles from within _inline_ plugins? That would be a nifty way to allow declaration of global styles from virtually anywhere in your app.

Something like this should work:

```js
tw(() => ({     
  ':global': { /* global styles */ },
}))

tw(css({     
  ':global': { /* global styles */ },
}))
```

With `tw.apply` this _should_ be possible:

```js
tw(() => ({     
  ':global': { 
   a: tw.apply`text-blue-500`
  },
}))

tw(css({     
  ':global': { 
   a: tw.apply`text-blue-500`
  },
}))
```

**Comment by tylerforesthauser** [comment]
That honestly _looks_ amazing but I am having no luck actually getting either of those to output anything at all...

**Comment by sastan** [comment]
Sorry. My bad. This depends on #84 and is not yet released. You could try the PR package. It's described in the PR

**Comment by sastan** [comment]
Your first example (the one that does not work) can not work as tw always returns a string of class names. You'll need to use tw.apply for this which is not yet merged or released. See the component PR and try PR package.

**Comment by tylerforesthauser** [comment]
No no it's all good. I've got the changes merged into my local repo and I'm using the package from there, so it includes your changes from #85 and the `component-api` branch. The selector actually works! Just not for nested `tw.apply` functions. 

My end goal is to declare global styles (specifically on the `html` element) from somewhere other than from `preflight`. 

I realize I entered my code incorrectly in my comment above but I, indeed, used `tw.apply`. 

So to clarify from my comment above, the following examples WORK:


```js
tw(
  css({
    ':global': {
      html: {
        backgroundColor: '#333',
      },
    },
  })
)
```

```js
tw(() => ({
  ':global': {
    html: {
      backgroundColor: '#333',
    },
  },
}))
```

```js
tw(({ theme }) => ({
  ':global': {
    html: {
      backgroundColor: theme('colors.gray.900'),
    },
  },
}))
```

And these examples, however, DO NOT WORK. 

```js
tw(() => ({
  ':global': {
    html: tw.apply('bg-gray-900 text-white'),
  },
}))
```

```js
tw(css({     
  ':global': { 
   html: tw.apply('bg-gray-900 text-white'),
  },
}))
```

**Comment by sastan** [comment]
This should work in #83 (7b446c9)

**Comment by sastan** [comment]
@tylerforesthauser This issue has two topics. Using tw.apply in `:global` and in preflight should now be fixed in #83 . Preflight merging should be a new issue discussing its impact. Could you please create one only for merging preflight styles.

**Comment by tylerforesthauser** [comment]
Done
